simulariumio/tests/test_agents_from_lists.py

```diff
@@ -0,0 +1,344 @@
+                    [0.0, 0.0, 0.0],
+                    [0.0, 0.0, 0.0],
+                ],
+                [
+                    [0.0, 0.0, 0.0],
+                    [-43.37181102, -13.41127423, -17.31316927],
+                    [0.0, 0.0, 0.0],
+                    [0.0, 0.0, 0.0],
+                ],
+                [
+                    [-24.91450698, -44.79360525, 13.32273796],
+                    [4.10861266, 43.86451151, 21.93697483],
+                    [0.0, 0.0, 0.0],
+                    [0.0, 0.0, 0.0],
+                ],
+            ],
+        ),
+        radii=np.array(
+            [
+                [8.38656327, 1.0, 1.0, 0.0],
+                [1.0, 6.69209780, 1.0, 1.0],
+                [8.91022619, 9.01379396, 1.0, 0.0],
+            ],
+        ),
+    )
+    result = AgentData.from_lists(list_data, 1.0)
+    assert expected_results == result
+
+
+def test_from_list_with_scale_factor():
+    times = [0.0, 0.1, 0.2]
+    n_agents = [3, 3, 3]
+    viz_types = [
+        [1000.0, 1001.0, 1001.0],
+        [1001.0, 1000.0, 1001.0],
+        [1000.0, 1000.0, 1001.0],
+    ]
+    unique_ids = [
+        [0.0, 1.0, 2.0],
+        [0.0, 1.0, 2.0],
+        [0.0, 1.0, 2.0],
+    ]
+    types = [
+        ["H", "A", "C"],
+        ["L", "D", "A"],
+        ["E", "Q", "K"],
+    ]
+    positions = [
+        [
+            [0.489610492, -2.981564851, 4.077254057],
+            [0.0, 0.0, 0.0],
+            [0.0, 0.0, 0.0],
+        ],
+        [
+            [0.0, 0.0, 0.0],
+            [-4.337181102, -1.341127423, -1.731316927],
+            [0.0, 0.0, 0.0],
+        ],
+        [
+            [-2.491450698, -4.479360525, 1.332273796],
+            [0.410861266, 4.386451151, 2.193697483],
+            [0.0, 0.0, 0.0],
+        ],
+    ]
+    radii = [
+        [0.838656327, 0.1, 0.1],
+        [0.1, 0.669209780, 0.1],
+        [0.891022619, 0.901379396, 0.1],
+    ]
+
+    list_data = AgentDataLists(
+        times=times,
+        n_agents=n_agents,
+        viz_types=viz_types,
+        unique_ids=unique_ids,
+        types=types,
+        positions=positions,
+        radii=radii,
+    )
+
+    scale_factor = 10.0
+    expected_results = AgentData(
+        times=np.array(times),
+        n_agents=np.array(n_agents),
+        viz_types=np.array(viz_types),
+        unique_ids=np.array(unique_ids),
+        types=types,
+        positions=scale_factor * np.array(positions),
+        radii=scale_factor * np.array(radii),
+    )
+
+    result = AgentData.from_lists(list_data, scale_factor)
+    assert expected_results == result
+
+
+def test_jagged_data_subpoints():
+    times = [0.0, 1.00001, 2.00001]
+    n_agents = [3, 3, 3]
+    viz_types = [
+        [1001.0, 1001.0, 1001.0],
+        [1001.0, 1001.0, 1001.0],
+        [1001.0, 1001.0, 1001.0],
+    ]
+    unique_ids = [[1.0, 2.0, 3.0], [1.0, 2.0, 3.0], [1.0, 2.0, 3.0]]
+    types = [["H", "A", "C"], ["L", "D", "A"], ["K", "K", "A"]]
+    positions = [
```

Comment:
where you're just filling in dummy data, could you use `.fill()` so there's just less input data to read?